x-pack/plugins/security_solution/public/common/components/event_details/table/use_action_cell_data_provider.ts

```diff
@@ -61,6 +63,10 @@ export const useActionCellDataProvider = ({
     let valueAsString: string = isString(value) ? value : `${values}`;
     if (fieldFromBrowserField == null) {
       stringifiedValues.push(valueAsString);
+      if (isThreatMatch) {
+        id = `threat-match-${contextId}-${field}-${value}-${eventId}`;
```

Comment:
Is this logic due to the potential for threat matched fields to be duplicated? If that's true, I think we can fix that higher up by ensuring that `e.g.` our `contextId` is unique for the match.

Reply:
so I believe this is the source of the timeline bug actually - this function is meant to return an `{ idList, stringValues}` object where the `idList` seems to contain the unique id of the draggable, and the `stringValues` is an array of values of that field that is being dragged. And I believe there is supposed to be a counterpart that matches the `idList` ids in the DOM which is relevant to a drop target, and that seems to be the missing piece, I went into a bit of a rabbit hole in there and I thought that I was probably missing something simple

Reply:
I added this piece because without it, idList is an empty array (as the threat intel fields are not included as a part of the id generation logic in the same file), and when the idList is empty, the timeline icon does not show up. This ties back to the question I asked on slack, regarding us not being able to pre-identify the threat match fields easily considering that the indicator match can happen on any field, which is why I added the isThreatMatch flag - but that might not exactly work for the reasons listed above (missing counterpart in the DOM)